MTM-46642 (#79)
SoftwareAG/c8y-releasenotes: content/release-10-15-0/announcements-10-15-0.md

Instead, a time-to-live configuration or retention rules can be used to remove expired measurements data from the Operational Store.

##### Breaking change in the Alarms/Events/Measurements API - unsupported query parameters will be rejected and required parameters will be introduced

> As of release 10.17+, it will no longer be possible to pass unsupported http query parameters, so also all request with wrongly typed parameters will be rejected.

Reviewer: ```suggestion
As of release 10.17+, it will no longer be possible to pass unsupported HTTP query parameters. This means that also all requests with wrongly typed parameters will be rejected.
```

> The change affects Alarms, Events, Measurements APIs.

Reviewer: ```suggestion
This change affects the alarms, events, and measurements APIs.
```

> Some query parameters will be required to prevent accidental deletion of too many objects when deleting, and to prevent transferring too much unrequited data when calling GET collections APIs.

Reviewer 2: I guess we need to further define the exact parameters that will be mandatory...

> Some query parameters will be required to prevent accidental deletion of too many objects when deleting, and to prevent transferring too much unrequited data when calling GET collections APIs.

Author: I've added the info about required parameters

> At least one parameter limiting the affected data e.g. `dateFrom`, `dateTo`, `type` should be provided.

Reviewer: ```suggestion
At least one parameter limiting the affected data, for example `dateFrom`, `dateTo`,  or `type`, should be provided.
```

Instead, a time-to-live configuration or retention rules can be used to remove expired measurements data from the Operational Store.

##### Breaking change in the Alarms, Events, Measurements APIs - unsupported query parameters will be rejected and required parameters will be introduced

As of release 10.17+, it will no longer be possible to pass unsupported HTTP query parameters. This means that also all requests with wrongly typed parameters will be rejected.
This change affects the Alarms, Events and Measurements APIs.

At least one query parameter limiting the affected data will be required to prevent accidental deletion of too many objects during a bulk delete operation.

> * `DELETE /alarm/alarms` will required at least one of the following parameter: `source`, `dateFrom`, `dateTo`, `createdFrom`, `createdTo`

Reviewer: ```suggestion
* `DELETE /alarm/alarms` will require at least one of the following parameters: `source`, `dateFrom`, `dateTo`, `createdFrom`, `createdTo`
```

> * `DELETE /alarm/alarms` will require at least one of the following parameters: `source`, `dateFrom`, `dateTo`, `createdFrom`, `createdTo`

Reviewer: ```suggestion
* `DELETE /alarm/alarms` requires at least one of the following parameters: `source`, `dateFrom`, `dateTo`, `createdFrom`, `createdTo`
```

> * `DELETE /event/events` will required at least one of the following parameter: `source`, `dateFrom`, `dateTo`, `createdFrom`, `createdTo`

Reviewer: ```suggestion
* `DELETE /event/events` will require at least one of the following parameters: `source`, `dateFrom`, `dateTo`, `createdFrom`, `createdTo`
```

> * `DELETE /event/events` will require at least one of the following parameters: `source`, `dateFrom`, `dateTo`, `createdFrom`, `createdTo`

Reviewer: ```suggestion
* `DELETE /event/events` requires at least one of the following parameters: `source`, `dateFrom`, `dateTo`, `createdFrom`, `createdTo`
```

> * `DELETE /measurements/measurement` will required at least one of the following parameter: `source`, `dateFrom`, `dateTo`

Reviewer: ```suggestion
* `DELETE /measurements/measurement` will require at least one of the following parameters: `source`, `dateFrom`, `dateTo`
```

> * `DELETE /measurements/measurement` will require at least one of the following parameters: `source`, `dateFrom`, `dateTo`

Reviewer: ```suggestion
* `DELETE /measurements/measurement` requires at least one of the following parameters: `source`, `dateFrom`, `dateTo`
```
